src/python/ensembl/io/genomio/seq_region/collection.py

```diff
@@ -0,0 +1,260 @@
+                self.seqs[name] = merged_seq
+
+    def _merge(self, source, destination):
+        """Merge a source dict in a destination dict (only add values or append to lists)."""
+        if not destination:
+            return source
+        for key, value in source.items():
+            if isinstance(value, list):
+                destination[key] += value
+            else:
+                destination[key] = value
+
+        return destination
+
+    @staticmethod
+    def make_seqregion_from_gbff(record_data: GBFFRecord) -> SeqRegionDict:
+        """Returns a seq_region dict extracted from a GBFF record."""
+        seqr: SeqRegionDict = {"length": len(record_data.record.seq)}
+
+        if record_data.is_circular():
+            seqr["circular"] = True
+
+        # Is there a genetic code defined?
+        codon_table = record_data.get_codon_table()
+        if codon_table is not None:
+            seqr["codon_table"] = codon_table
+
+        # Is it an organelle?
+        location = record_data.get_organelle()
+        if location is not None:
+            seqr["location"] = location
+
+        # Is there a comment stating the Genbank record this is based on?
+        genbank_id = record_data.get_genbank_id()
+        if genbank_id is not None:
+            seqr["synonyms"] = [{"source": "INSDC", "name": genbank_id}]
+
+        return seqr
+
+    def from_report(self, report_path: Path, is_refseq: bool = False) -> None:
+        """Store seq_regions extracted from an INSDC assembly report file.
+        If a seq_region with the same id exists in the collection, it will be replaced.
+
+        Args:
+            report_path: Path to the sequence regions report file.
+            is_refseq: True if the source of the report is RefSeq, false if INSDC.
+
+        """
+        report = ReportRecord(report_path)
+        for seq_data in report.reader:
+            new_seq = self.make_seq_region_from_report(seq_data, is_refseq)
+            name = new_seq["name"]
+            merged_seq = self._merge(new_seq, self.seqs.get(name, {}))
+            self.seqs[name] = merged_seq
+
+    @staticmethod
+    def make_seq_region_from_report(
+        seq_data: dict,
+        is_refseq: bool,
+        synonym_map: dict[str, str] | None = None,
+        molecule_location: dict[str, str] | None = None,
+    ) -> SeqRegionDict:
+        """Returns a sequence region from the information provided.
+
+        An empty sequence region will be returned if not accession information is found.
+
+        Args:
+            data: a dict from the report representing one line, where the key is the column name.
+            is_refseq: True if the source is RefSeq, false if INSDC.
+            synonym_map: Map of INSDC report column names to sequence region field names.
+            molecule_location: Map of sequence type to SO location.
+
+        Raises:
+            KeyError: If the sequence location is not recognised.
```

Comment:
```suggestion
```
Based on the code of this function, `KeyError` is not an exception that you intentionally raise, so it should not be part of the docstring.